src/syft/core/common/module.py

```diff
@@ -77,3 +83,81 @@ def cpu(self) -> "Module":
         for _, module in self.modules.items():
             module.cpu()
         return self
+
+    # zero them so we know they are copied
+    def zero_layers(self) -> None:
+        for _, m in self.modules.items():
+            if hasattr(m, "weight"):
+                m.weight.requires_grad_(False).zero_()
+            if hasattr(m, "bias"):
+                m.bias.requires_grad_(False).zero_()
+
+    # easy way to check the weights have changed
+    def sum_layers(self) -> None:
+        for n, m in self.modules.items():
+            if hasattr(m, "state_dict"):
+                for k, v in m.state_dict().items():
+                    if hasattr(v, "sum"):
+                        s = v.sum().item()
+                        log = f"> Layer {n} sum({k}): {s}"
+                        print(log)
+                        logger.debug(log)
+
+    # copy the weights, there are blocking requests so make sure to respond
+    def copy_remote_state(
+        self,
+        remote_model: Any,
+        request_name: str,
+        reason: str,
+        timeout_secs: int,
+        delete_obj: bool = False,
+        skip_layers: List[str] = [],
+    ) -> None:
+        # loop over models module pointers
+        for n, m in remote_model.modules.items():
+            try:
+                if n in skip_layers:
+                    log = f"> Skipping: {n}"
+                    print(log)
```

Comment:
Q: Shouldn't we remove the prints? and simply leave the  ```logger.debug```
We can change the debug level from ```__init__.py```

Reply:
I haven't gotten the logging system to filter the print statements in a way that im happy with yet, so I have some areas where the printing is explicitly designed to appear in the notebook, but we need to figure out a solid way to make this consistent. I think what we need to do is add our own custom logging levels and use those so that we arent restricted by the normal order of logging importance. I have made a ticket to work on this independently. https://github.com/OpenMined/PySyft/issues/4755